integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala

```diff
@@ -156,4 +158,132 @@ object DataLoadProcessBuilderOnSpark {
       Array((uniqueLoadStatusId, (loadMetadataDetails, executionErrors)))
     }
   }
+
+  /**
+   * 1. range partition the whole input data
+   * 2. for each range, sort the data and writ it to CarbonData files
+   */
+  def loadDataUsingRangeSort(
+      sparkSession: SparkSession,
+      dataFrame: Option[DataFrame],
+      model: CarbonLoadModel,
+      hadoopConf: Configuration): Array[(String, (LoadMetadataDetails, ExecutionErrors))] = {
+    val originRDD = if (dataFrame.isDefined) {
```

Comment:
This method has too much of the same code  as loadDataUsingGlobalSort， I recommend refactoring these two methods

Reply:
better, but after refactoring,  the code logic is not clear. Now, these two flows already reuse the process steps.